File: src/main/java/com/cfgglobal/generator/entity/CodeProject.java
package com.cfgglobal.generator.entity;

import com.cfgglobal.generator.core.ScriptHelper;
import com.cfgglobal.generator.core.TemplateHelper;
import lombok.Data;

import java.io.File;
import java.util.List;

@Data
public class CodeProject {
    private static final String FS = File.separator;

    String packageName;

    String templatePath;

    String targetPath;

    ScriptHelper scriptHelper;

    TemplateHelper templateEngine;

    List<CodeEntity> entities;

    List<Class> utilClasses;


}

File: src/main/kotlin/com/cfgglobal/generator/entity/CodeProject.kt
package com.cfgglobal.generator.entity

import com.cfgglobal.generator.core.ScriptHelper
import com.cfgglobal.generator.core.TemplateHelper
import com.cfgglobal.generator.task.service.TaskService
import java.io.File

data class CodeProject(
        var packageName: String,

        var templatePath: String,

        var scriptHelper: ScriptHelper,

        var templateEngine: TemplateHelper,

        var entities: List<CodeEntity> = listOf(),

        var utilClasses: List<Class<*>> = listOf(),

        var apiTasks: List<Task> = listOf(),

        var apiTargetPath: String,

        var uiTargetPath: String,

        var uiTasks: List<Task> = listOf()

){
    fun generate(){
        this.apiTasks.forEach {
            TaskService.processTask(this,it)
        }

        this.uiTasks.forEach {
            TaskService.processTask(this,it)
        }
    }
}

